Add org-notifications recipe

doppelc opened the pull request:
### Brief summary of what the package does

This package creates system notifications for org-agenda items with timestamps.

- Notify a set couple of minutes before the timestamp
- Choose which agenda or non-agenda files to check or both
- Optional sound on notifications
- Exclude and include specific agenda tags
- Notifies for repeated items

### Direct link to the package repository

https://github.com/doppelc/org-notifications

### Your association with the package

Maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Thanks for this.  I may be wrong about these comparisons, but have you had a chance to see how this differs from org-alert and org-wild-notifier (or if there are possible collaboration opportunities)?

Some quick suggestions:
- The default value for `org-notifications-non-agenda-file` seems like it may be inviting trouble.  Does a user have to set this correctly before using the package?  Could the list just be empty to start?
- org-notifications.el#L134: It's safer to sharp-quote function names; use `#'`
- org-notifications.el#L218: It's safer to sharp-quote function names; use `#'`
- Very minor, but when I see a function like `org-notifications--time-to-hhmm` just used in one place, I always think it's better to inline it
- Some of your indentation is inconsistent (Emacs should be basically handling this for you 100%) and some lines extend to 170 characters, making things a little harder than they need to be to parse
- Not a blocker, but it will be less ambiguous for automatic detection tools if you add complete license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to org-notifications.el

doppelc commented:
@riscy Thank you for the review, I really appreciate it!

org-alert: Notifies every x minute regardless of time of the day or time to the scheduled entry. The maintainer is not active anymore.

org-wild-notifier: It seems to work somewhat similarly and the maintainer seems to be active. While it does allow you to set several notifications for each entry, it does not, it seems to me, allow you to:
- Notify for repeat items
- Select which agenda files to check
- Have sounds for notifications
- Set package specific notification style

Below changes are made in [commit.](https://github.com/doppelc/org-notifications/commit/47802d64340ab1ff2c5f2a85e395b15cef504d86)

> The default value for org-notifications-non-agenda-file seems like it may be inviting trouble. Does a user have to set this correctly before using the package? Could the list just be empty to start?

This variable is only used if `org-notifications-which-agenda` files is set appropriately. I've modified the docstring to reflect this. Even if  `org-notifications-which-agenda` is deliberately set to use `org-notifications-non-agenda` file and if it's set incorrectly, it won't cause a problem as it will be ignored. This way it's easy for users to see how it should be set, I think.

> org-notifications.el#L134: It's safer to sharp-quote function names; use #'

Fixed.
> org-notifications.el#L218: It's safer to sharp-quote function names; use #'

Fixed.

> Very minor, but when I see a function like org-notifications--time-to-hhmm just used in one place, I always think it's better to inline it

Sound feed back, thank you. Fixed.

> Some of your indentation is inconsistent (Emacs should be basically handling this for you 100%) and some lines extend to 170 characters, making things a little harder than they need to be to parse

Fixed.

> Not a blocker, but it will be less ambiguous for automatic detection tools if you add complete license boilerplate or an SPDX-License-Identifier to org-notifications.el

Fixed.

riscy commented:
> org-wild-notifier: It seems to work somewhat similarly and the maintainer seems to be active. While it does allow you to set several notifications for each entry, it does not ...

I'm not prepared to comment on how difficult or how compatible this list of features is with org-wild-notifier, but I'm assuming you don't see an opportunity to talk to that package's maintainer about adding these?

This isn't a blocker by the way, but since you're a first-time contributor it bears mentioning that we try to push toward cooperation on existing packages when possible.

doppelc commented:
Thank you, I'll keep it in mind. It seems someone has already reported it as not working and made an attempt at repeated items, though it's been 6 months/a year without any feedback.

riscy commented:
This looks good to me.  Welcome to MELPA! :)
